### DISCUSSION TITLE
Go To Reference should exclude current item

### DISCUSSION THREAD

**Discussion by NoelAbrahams** [microsoft/vscode-discussions#778] [Extension Development QnA] [answered]
Hi, I'm raising this on behalf of the developer for the CSS Navigation extension. 

The problem is when go-to-reference is being invoked, the developer is getting the following results:

![image](https://github.com/microsoft/vscode-discussions/assets/1106823/d0ed7308-8660-4317-940b-d3ff97ab14a3)

It looks like VSCode automatically adds the current file to the list of discovered files. The current file is pointless because that's where the user is at the moment.

Should this be logged as a bug in VSCode? Or is there something that the extension developer can do?

Thanks
Noel

Link to the original issue: https://github.com/pucelle/vscode-css-navigation/issues/86

**Comment by burkeholland** [comment, marked as answer]
I added a comment to that issue. It looks to me like the developer is implementing `findReferences` in the extension, so they should be able to check for the current file and ignore it there. https://github.com/pucelle/vscode-css-navigation/blob/d86b5c0b3112012e11659355aa028bbec7b96ae3/server/src/server.ts#L373C2-L375C47
